test/common/upstream/eds_speed_test.cc

```diff
@@ -174,49 +173,45 @@ class EdsSpeedTest {
-                                       Envoy::Logger::Logger::DEFAULT_LOG_FORMAT, lock, false);
-  for (auto _ : state) {
-    Envoy::Upstream::EdsSpeedTest speed_test(state, state.range(0));
-    // if we've been instructed to skip tests, only run once no matter the argument:
-    uint32_t endpoints = skipExpensiveBenchmarks() ? 1 : state.range(2);
-
-    speed_test.priorityAndLocalityWeightedHelper(state.range(1), endpoints, true);
+  Envoy::Upstream::EdsSpeedTest speed_test(state, state.range(0));
+  // if we've been instructed to skip tests, only run once no matter the argument:
+  const uint32_t num_endpoints = skipExpensiveBenchmarks() ? 1 : state.range(2);
+  for (auto _ : state) { // NOLINT(clang-analyzer-deadcode.DeadStores)
+    state.PauseTiming();
+
+    speed_test.priorityAndLocalityWeightedHelper(state.range(1), num_endpoints, true);
+    state.ResumeTiming();
   }
 }
 
 BENCHMARK(priorityAndLocalityWeighted)
-    ->Ranges({{false, true}, {false, true}, {1, 100000}})
+    ->Ranges({{false, true}, {false, true}, {64, 100000}})
     ->Unit(benchmark::kMillisecond);
 
 static void duplicateUpdate(State& state) {
-  Envoy::Thread::MutexBasicLockable lock;
-  Envoy::Logger::Context logging_state(spdlog::level::warn,
-                                       Envoy::Logger::Logger::DEFAULT_LOG_FORMAT, lock, false);
-
-  for (auto _ : state) {
-    Envoy::Upstream::EdsSpeedTest speed_test(state, false);
-    uint32_t endpoints = skipExpensiveBenchmarks() ? 1 : state.range(0);
-
-    speed_test.priorityAndLocalityWeightedHelper(true, endpoints, true);
-    speed_test.priorityAndLocalityWeightedHelper(true, endpoints, true);
+  Envoy::Upstream::EdsSpeedTest speed_test(state, false);
+  const uint32_t num_endpoints = skipExpensiveBenchmarks() ? 1 : state.range(0);
+  for (auto _ : state) { // NOLINT(clang-analyzer-deadcode.DeadStores)
+    state.PauseTiming();
+
+    speed_test.priorityAndLocalityWeightedHelper(true, num_endpoints, true);
+    speed_test.priorityAndLocalityWeightedHelper(true, num_endpoints, true);
+    state.ResumeTiming();
   }
 }
 
-BENCHMARK(duplicateUpdate)->Range(1, 100000)->Unit(benchmark::kMillisecond);
+BENCHMARK(duplicateUpdate)->Range(64, 100000)->Unit(benchmark::kMillisecond);
 
 static void healthOnlyUpdate(State& state) {
-  Envoy::Thread::MutexBasicLockable lock;
-  Envoy::Logger::Context logging_state(spdlog::level::warn,
-                                       Envoy::Logger::Logger::DEFAULT_LOG_FORMAT, lock, false);
-  for (auto _ : state) {
-    Envoy::Upstream::EdsSpeedTest speed_test(state, false);
-    uint32_t endpoints = skipExpensiveBenchmarks() ? 1 : state.range(0);
-
-    speed_test.priorityAndLocalityWeightedHelper(true, endpoints, true);
-    speed_test.priorityAndLocalityWeightedHelper(true, endpoints, false);
+  Envoy::Upstream::EdsSpeedTest speed_test(state, false);
+  const uint32_t num_endpoints = skipExpensiveBenchmarks() ? 1 : state.range(0);
```

Comment:
How come you're changing `eds_speed_test.cc` as well?

Reply:
This (should be) a port of jmarantz's style comments from ```cds_speed_test```